components/speedreader/rust/ffi/BUILD.gn

```diff
@@ -0,0 +1,69 @@
+# Copyright 2020 The Brave Authors. All rights reserved.
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this file,
+# You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import("config.gni")
+import("//brave/build/cargo.gni")
+import("//brave/test/testing.gni")
+
+config("external_config") {
+  include_dirs = [ "include" ]
+  defines = [
+    "NO_CXXEXCEPTIONS=1"
+  ]
+}
+
+source_set("speedreader_ffi") {
+  public_configs = [ ":external_config" ]
+
+  sources = [
+    "src/speedreader.cc",
+    "include/speedreader.h"
+  ]
+
+  deps = [
+    "//base",
+    ":speedreader"
+  ]
+
+  libs = [ speedreader_lib_path ]
+
+  if (is_mac || is_ios ) {
+    libs += [
+      "Security.framework"
+    ]
+  }
+}
+
+if (speedreader_shared_lib_install_name != "") {
+  fix_testing_install_name("test_speedreader_install_name") {
+    current_path = speedreader_shared_lib_install_name
+    new_path = "@executable_path/" + rebase_path(speedreader_lib_path, root_build_dir)
+  }
+} else {
+    group("test_speedreader_install_name") {}
+}
+
+cargo_build("speedreader") {
+  inputs = [
+    "build.rs",
+    "Cargo.toml",
+    "cbindgen.toml",
+    "include/speedreader_ffi.h",
+    "src/charbuf.rs",
+    "src/errors.rs",
+    "src/lib.rs",
+    "src/speedreader.rs",
+    "../src/src/whitelist.rs",
```

Comment:
also why is there a bunch of stuff in `src/src` instead of just `src`?

Reply:
wanting to keep the FFI part of the library separate from the logic, we discussed having `brave/components/speedreader/rust/src` and `brave/components/speedreader/rust/ffi` and [Cargo package convention](https://doc.rust-lang.org/cargo/guide/project-layout.html) puts sources within `src`.

Happy to rename (src -> lib, so it would be lib/src?), but doesn't seem useful to just merge them